### ISSUE TITLE
[Proposal] Config builder

### ISSUE THREAD

**Issue by Siemienik** [Siemienik/XToolset#31]
# The Config Builder - still DRAFT

_Please do not start until it is a draft_

## Place of live

`./src/configBuilder/`

## Sample usages
```js

const cfg = ListCfgBuilder.createList()
  .worksheet('asd')
  .column('key', 1 ,integerMapper)
  .column('key2' 2, splitMapper.separator(':'))
  .toConfig(); // or .build() - or event better, put this cfg directly into getAllItems() 

// and
const cfg2 =ListCfgBuilder.from(cfg)
  .worksheet('second')
  .getConfig();


// json support
const asJson = CfgBuilder.from(cfg).toJson();

const fromJson = CfgBuilder.from(asJson, {type: 'json'}).toConfig();
```

## TODO

* [ ] ListVerticalConfigBuilder
* [ ] SingleObjectConfigBuilder
* [ ] Research required: ConfigBuilder for both
* [ ] toJson/fromJson (ConfigWriter/ConfigReader)
    * [ ] string representation for mappers

## Draft's todo:

* [ ] analysis discriminator for reading config
* [ ] become sure about APIs
* [ ] create issues for each TODO Point
* [ ] consider about moving ConfigWriter & Reader out of scope of this issue
* [ ] write about motivations to add it into project

**Comment by Metastasis** [comment]
@Siemienik 
Could you add a little description?

**Comment by Siemienik** [comment]
I'm currently under analysing it and I will describe it after 2.3.2 version released - what I plan to do at 18th or 19th oct2020.

as introduction, I imagine like that

```js
const cfg = CfgBuilder.createList()
  .worksheet('asd')
  .addColumn('key',{col:1,row:1},integerMapper)
  .addColumn(...)
  .getConfig(); // or .build() - or event better, put this cfg directly into getAllItems() 

// and
const cfg2 =CfgBuilder.from(cfg)
  .worksheet('second')
  .getConfig();
```


I plan to separate this for two issues - one per import strategy.

This is only pre-written idea, I plan to describe it in more detail 😄

**Comment by Siemienik** [comment]
I've done research, however I wait for #62 and #63 to be 100% sure that everything is included.
